### ISSUE TITLE
Disabling private browsing via Policy is not working

### ISSUE THREAD

**Issue by Daemach** [brave/brave-browser#5902]
I want my kids to use Brave because of the tracker blocking, but do not want them using private windows.  Keep honest people honest, as my grandfather used to say.  How do I disable private and tor browsing via the registry, group policy or some other mechanism that they don't have access to?  I can disable private browsing in Edge, Firefox and Chrome - please don't make me uninstall Brave...

**Comment by bsclifton** [comment]
We should be inheriting this behavior from Chrome/Chromium... likely the registry keys are different (on Windows)

@Daemach what platform are you on?

I did find this for Windows, which we (at Brave) could potentially test to see what steps differ on Brave vs Chrome:
https://community.spiceworks.com/how_to/125579-how-to-disable-chrome-incognito-mode

**Comment by Daemach** [comment]
The machine in question is Windows 10.  I used the registry - like in this article: https://www.technipages.com/chrome-disable-incognito-mode.  Chrome picked it up but Brave didn't.  It may have had processes running in the background though, and I probably should have rebooted before I posted this.  I do need to make sure that it will be blocking private browsing via Tor as well.  

I appreciate your looking into this!

**Comment by Daemach** [comment]
@bsclifton I rebooted the machine and it's not picking up that registry setting. https://www.screencast.com/t/pn0rfpieVM  Inheriting the Chrome registry key would be the preferred method since it works for Chrome already.   Or maybe you could add another registry key specifically for Brave in the same location (localmachine/software/profiles) if you want to keep it separate.  Really, though, if someone doesn't want private browsing on Chrome they probably don't want it on any of their browsers so if they go through the trouble of editing the registry you would probably be safe in inheriting that setting.

**Comment by rebron** [comment]
@Brave-Matt Can we create an article for this one and put up on Help Center?

**Comment by Daemach** [comment]
Once we're able to disable private browsing that would be a great idea.

**Comment by Daemach** [comment]
@bsclifton Any movement on this?  This is an important issue.

**Comment by bsclifton** [comment]
Thanks for pinging again, @Daemach - I dug in and verified there is a problem.

Basically, for a 64 bit machine, you should be able to block Private Browsing by creating the following DWORD key:
`IncognitoModeAvailability`: `1`
under:
`HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Policies\BraveSoftware\Brave`

I verified that even after a restart, this did not work. I verified the policy name matches the expected string from the Chromium code and also verified that `1` is the expected value (0 = allow private browsing (default), 1 = block private browsing, 2 = ONLY use private browsing)

Will need to investigate why this isn't working...

**Comment by bsclifton** [comment]
The path called out above should work great once https://github.com/brave/brave-core/pull/3692 is merged (since it provides the proper `CHROMIUM_POLICY_KEY`)

In the meantime, I verified it DOES disable as expected if you create the `IncognitoModeAvailability` value with `1` under the following key:
`HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Policies\Chromium`

@Daemach can you try setting BOTH locations? (so then, when we update the path, you won't have to change it again?). Let us know if that works! 😄

**Comment by bsclifton** [comment]
Confirmed that this does work (see above note about which keys to set). Any of the policy settings captured in templates (see https://support.google.com/chrome/a/answer/187202?hl=en) work as expected (I tried several, including forcing extension install)

@Brave-Matt will be working on a help article to make this a little easier to understand. Let us know if the above works, @Daemach! 😄

**Comment by RJMFROMMINN** [comment]
This is not working for me. I've tried editing the registry under both \Brave and \Chromium, to no avail.

**Comment by bsclifton** [comment]
@RJMFROMMINN you're trying to toggle the `IncognitoModeAvailability` one specifically? Or another value?

**Comment by RJMFROMMINN** [comment]
No, I am trying to toggle the IncognitoModeAvailability specifically. I open up Brave, and I can use Private Mode.

**Comment by bsclifton** [comment]
@RJMFROMMINN what version of Brave are you running? Also, try this path instead (the non-64 bit version; I think they're linked anyways):
`HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Chromium`

**Comment by RJMFROMMINN** [comment]
I had tried this about two weeks ago. Tried again earlier this afternoon. Yes, I restarted the computer as well. Yes, I have also added the DWORD code to HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Chromium 

I have it in all three places in the registry editor. 

Windows 10 Home Version 1903 64-bit.

**Comment by bsclifton** [comment]
@RJMFROMMINN does it work with Chrome? The path there would be `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Google\Chrome\`

I'm looking for details about support - I know the group policy tool isn't supported on Windows Home editions. The registry key approach should still work though... will update if I find more details

**Comment by RJMFROMMINN** [comment]
Yes, it works with Chrome. I've had private browsing disabled from both Chrome and Edge for some time now. 

When I pulled up brave://policy, it lists Incognito Mode Availability as "1". 

![image](https://user-images.githubusercontent.com/58236021/69734551-c6b47000-10f4-11ea-973e-3a6d29e697db.png)

Yet when I go to the menu, I can still open a "New Private Window with Tor."

**Comment by KieranP** [comment]
I'm on a mac. Using this worked to disable private browsing:

```
defaults write com.brave.Browser IncognitoModeAvailability -integer 1
```

However the option for private window with tor still exists, but when clicked, it crashes the browser.

**Comment by bsclifton** [comment]
@KieranP great, glad that disabled regular private windows! The Tor windows have a distinct setting which only was added recently: `TorDisabled`. 

Unfortunately, this is only on version 1.3 (hitting release channel Feb 4th) and newer

When this does go live, you should be able to disable using:
```
defaults write com.brave.Browser TorDisabled -integer 1
```

**Comment by Daemach** [comment]
Messy. Why not another registry setting?  Or better, a single setting to disable all private browsing?  And where would you use this anyway?

iPhone. iTypos. iApologize...

________________________________

**Comment by bsclifton** [comment]
@Daemach for Windows, it would be a registry setting 😄 @KieranP is on macOS

For Windows, the key would be: 
- `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave`
- Create new DWORD `TorDisabled`
- Set to 1

**Comment by RJMFROMMINN** [comment]
Hey, BSClifton, that did it. I added the TorDisabled code to my registry, and now Brave crashes when I try to go to a private window with Tor. 

Thanks :)

**Comment by bsclifton** [comment]
@RJMFROMMINN awesome! Thanks for confirming 😄

**Comment by Daemach** [comment]
This fix works well.  Thank you very much for your prompt response and for
seeing it through.

Please also add a registry key or group policy setting to disable clearing
browser history.  I created a new issue for that request here:
https://github.com/brave/brave-browser/issues/7954

Keeping honest people honest...

And in an enterprise setting, clearing browser history in situations such
as intellectual property theft, is tantamount to destruction of evidence.
Enterprise environments need more control over these settings, as do
parents.

**Comment by sidhreddy** [comment]
@bsclifton I am on macos. I tried the below on 1.3.118 and Brave crashes upon "New Private Window with Tor" which works for me for what I wanted to make sure I do not even accidentally use Tor. But I still see the "New Private Window with Tor". Do you if there is work being done to Gray out the option entirely??

> ```
> defaults write com.brave.Browser TorDisabled -integer 1
> ```

**Comment by gusmacaulay** [comment]
How would I do this on linux?

**Comment by zpsaras7** [comment]
@sidhreddy I also experienced the same issues trying to disable Tor and WebTorrent functionality via policy on MacOS. 

I wonder if there's a way to 'force' settings (specifically brave-packaged extensions) using plist files to make Brave more manageable for enterprises similar to the way Chrome/Chromium can enforce policies.

**Comment by mjplacroix** [comment]
Can't thank you enough for tackling this issue! Disabling this was a _big_ help.

**Comment by illphate** [comment]
> How would I do this on linux?

Can this thread be reopened? I need to know how to disable the tor browser for linux too
